### ISSUE TITLE
Vulkan: ImmediateGeometry3D is not visible

### ISSUE THREAD

**Issue by nightrobin** [godotengine/godot#39451]
**Godot version:**
4.0.dev Commit: ffbea8a

**OS/device including version:**
Windows 10 - 64 Bit

**Issue description:**
ImmediateGeometry3D is not visible when I try to draw a line
I **haven't** encoutered this issue on Version 3.2.1 stable

**Steps to reproduce:**
On function _ready()

```gdscript
$ImmediateGeometry3D.clear()
$ImmediateGeometry3D.begin(Mesh.PRIMITIVE_LINES, null)
$ImmediateGeometry3D.add_vertex(Vector3(1, 2, 1))
$ImmediateGeometry3D.add_vertex(Vector3(3, 2, 3))
$ImmediateGeometry3D.end()
```

**Minimal reproduction project:**
[Draw.zip](https://github.com/godotengine/godot/files/4762553/Draw.zip)

**Comment by clayjohn** [comment]
ImmediateGeometry probably hasn't been implemented yet.

**Comment by nightrobin** [comment]
@clayjohn What class/object I can use as a workaround for the meantime that I can use to draw lines? :)

**Comment by clayjohn** [comment]
You may be able to use ArrayMesh. Take a look at [this doc](https://docs.godotengine.org/en/latest/tutorials/content/procedural_geometry/index.html) for alternatives. 

Keep in mind that 4.0 is unstable and isn't even in an alpha stage yet. You can't expect to be able to do everything in it yet. 

You really should be using 3.2 if you want to work on any projects.

**Comment by nightrobin** [comment]
Thanks!

**Comment by nightrobin** [comment]
@Calinou I encounter this error even when I choose GLES2 in the Project Settings.

**Comment by akien-mga** [comment]
There's no GLES2 backend in the current `master` branch.

Again, if you're not familiar with the current state of the development branch, it's is **strongly** recommended that you stick to the stable 3.2 branch. Godot 4.0 is under heavy development and definitely not usable in production, nor supported as such.

**Comment by nightrobin** [comment]
Noted on this! 😅

**Comment by Calinou** [comment]
This issue is still relevant - we should reimplement ImmediateGeometry3D in Vulkan.

**Comment by TokageItLab** [comment]
Ah, I was looking for this issue and it has caused some similar issues to be posted. Some classes are using ImmediateGeometry and all of them do not work correctly in 4.0.
- #45304
- #47901
- #48389

**Comment by akien-mga** [comment]
Fixed by #50014.
